packages/react-reconciler/src/ReactFiberLane.new.js

```diff
@@ -751,6 +754,30 @@ export function markRootFinished(root: FiberRoot, remainingLanes: Lanes) {
     eventTimes[index] = NoTimestamp;
     expirationTimes[index] = NoTimestamp;
 
+    if (enableCache) {
+      // Subsequent loads in this lane should use a fresh cache.
+      // TODO: If a cache is no longer associated with any lane, we should issue
+      // an abort signal.
```

Comment:
This todo won't work because there can still be pending loads that will end up being used later after a state change reveals them. In fact, that's a key feature that we can preload other branches so they don't unexpectedly load later when they're actually rendered.

Instead, we can issue abort signal when a refresh would destroy this cache. We can do that eagerly in the imperative refresh call.

This still leaves cases when the signal is not invoked because we don't have a cleanup phase that can capture it all, but that might be fine because it's only an optimization - not a guarantee.

Reply:
> Instead, we can issue abort signal when a refresh would destroy this cache. We can do that eagerly in the imperative refresh call.

Wouldn't that suffer from the same problem, though, since multiple trees can share the same cache?

Reply:
You would need something like the subscription logic I was using to track siblings to know for sure that none of the requests are needed.

Reply:
You can ref count. You might not hit zero unless all are replaced instead of dropped.